### PR TITLE
fix(express): Ensure we document error handler positioning

### PR DESCRIPTION
**Pull request by mydea**
Fixes https://github.com/getsentry/sentry-javascript/issues/12113

### CODE REVIEW

File: platform-includes/getting-started-add-top-level-import/javascript.express.mdx [hunk 1]
```diff
@@ -8,7 +8,8 @@ const app = express();
 
 // Add your routes, etc.
 
-// Add this after all routes and other middlewares are defined
+// Add this after all routes,
+// but before any and other middlewares are defined
```
Comment (Lms24) [inline platform-includes/getting-started-add-top-level-import/javascript.express.mdx:12]: m: I think "middlewares" is a bit misleading as I'm fairly sure people also define middleware before they register their routes, no?

```suggestion
// but before any error handlers are defined
```

Comment (vivianyentran) [reply platform-includes/getting-started-add-top-level-import/javascript.express.mdx:12]: Would "error-handling middleware" work?

Comment (mydea) [reply platform-includes/getting-started-add-top-level-import/javascript.express.mdx:12]: I'll go with this 👍

**Review by AbhiPrasad — APPROVED** [review]
Because I just got bit with this lol

**Comment by pckilgore** [comment]
How would you recommend instrumenting more complicated router setups given the "before error middleware, but after routes" constraint?

For example:

```js
const app = express()
app.use(parserMiddlewares)
app.use(authorizationMiddleware) 

const apiRouter = Router()         // serves JSON for AJAX
apiRouter.all("/resource")
....
// < --- Position 1
apiRouter.use(errorMiddleware).     // serves JSON error responses.

app.use("/api", apiRouter) 

app.use(siteRouter)                 // serves HTML templates
// < --- Position 1
app.use(errorMiddleware)            // serves text/html error responses
```

Call `Sentry.setupExpressErrorHandler(app);` twice, once in each position?

**Comment by mydea** [comment]
Hey,

in this case, you need to add `setupExpressErrorHandler(app)` before each error middleware:

```js
const app = express()
app.use(parserMiddlewares)
app.use(authorizationMiddleware) 

const apiRouter = Router()         // serves JSON for AJAX
apiRouter.all("/resource")
....
// < --- Position 1
Sentry.setupExpressErrorHandler(apiRouter);
apiRouter.use(jsonErrorMiddleware).     // serves JSON error responses.

app.use("/api", apiRouter) 

app.use(siteRouter)                 // serves HTML templates
// < --- Position 1
Sentry.setupExpressErrorHandler(app);
app.use(htmlErrorMiddleware)            // serves text/html error responses
```

I think this should work! Really we just need to get the error before it is handled by some other middleware 😅

**Pull request merged**